File: parent/template/src/main/java/com/ysc/feign/Master.java
package com.ysc.feign;

import java.text.ParseException;
import java.util.List;
import java.util.Map;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;

import com.ysc.entity.Food;
import com.ysc.util.Result;

@FeignClient(value = "master")
public interface Master {

	@PostMapping("/getFlow")
	public Map<String,Object> get7DaysFlow() throws ParseException;
	
	@PostMapping(value = "/getFoods")
	public Map<String,Object> getFoods(@RequestParam(value="page",required = false) Integer page,
			@RequestParam(value="rows",required = false) Integer rows);
	
	@RequestMapping(value = "/edit",method = {RequestMethod.POST},produces = { "application/json;charset=UTF-8" })
	public Result editFood(@RequestParam("id") Integer id,
			@RequestParam("name") String name,@RequestParam("describe") String describe,
			@RequestParam("price") Double price);
	
	@PostMapping(value= "/delete")
	public Result deleteFood(@RequestParam("id") Integer id);
}

File: front/service/src/api/master.js
import request from '@/utils/request'

export function getComment() {
  return request({
    url: '/getComment', // 这个接口是为了服务端算一共有多少留言数的   客户端还有一个查询特定排序版的getComments
    method: 'post'
  })
}

export function getFlow() {
  return request({
    url: '/get7DaysFlow',
    method: 'post'
  })
}

export function getFoodList(page, rows) {
  return request({
    url: '/getFoodList',
    method: 'post',
    params: {
      page,
      rows
    }
  })
}

export function editFood(id, name, describe, price) {
  return request({
    url: '/editFood',
    method: 'post',
    params: {
      id,
      name,
      describe,
      price
    }
  })
}

export function deleteFood(id) {
  return request({
    url: '/deleteFood',
    method: 'post',
    params: {
      id
    }
  })
}
